Crash when generating iOS plugin

### Environment

node version: 16.7.1


### Is used with npx or as a local dependency?

- [X] npx
- [ ] local dependency

### Affected commands

- [ ] android
- [X] ios

### Current behavior

App crashes after generating plugin and running the app

https://github.com/mrousavy/react-native-vision-camera/issues/1307
https://github.com/mrousavy/react-native-vision-camera/issues/1295

### Expected behavior

App does not crash

### Reproduction repository link

I already reproduced it

> Receiving following crash when I run `npx vision-camera-plugin-builder ios`
> 
> 
> 
> ```shell
> 
> npm ERR! cb.apply is not a function
> 
> 
> 
> npm ERR! A complete log of this run can be found in:
> 
> npm ERR!     <directory>/.npm/_logs/2022-10-28T03_11_05_574Z-debug.log
> 
> Install for [ 'vision-camera-plugin-builder@latest' ] failed with code 1
> 
> ```
> 
> 
> 
> Here is log.
> 
> 
> 
> ```shell
> 
> 0 info it worked if it ends with ok
> 
> 1 verbose cli [
> 
> 1 verbose cli   '<directory>/.nvm/versions/node/v14.17.5/bin/node',
> 
> 1 verbose cli   '<directory>/.nvm/versions/node/v14.17.5/lib/node_modules/npx/node_modules/npm/bin/npm-cli.js',
> 
> 1 verbose cli   'install',
> 
> 1 verbose cli   'vision-camera-plugin-builder@latest',
> 
> 1 verbose cli   '--global',
> 
> 1 verbose cli   '--prefix',
> 
> 1 verbose cli   '<directory>/.npm/_npx/73253',
> 
> 1 verbose cli   '--loglevel',
> 
> 1 verbose cli   'error',
> 
> 1 verbose cli   '--json'
> 
> 1 verbose cli ]
> 
> 2 info using npm@5.1.0
> 
> 3 info using node@v14.17.5
> 
> 4 verbose npm-session e0aa96245e8dd287
> 
> 5 silly install loadCurrentTree
> 
> 6 silly install readGlobalPackageData
> 
> 7 http fetch GET 200 https://registry.npmjs.org/vision-camera-plugin-builder 29ms (from cache)
> 
> 8 silly pacote tag manifest for vision-camera-plugin-builder@latest fetched in 46ms
> 
> 9 verbose stack TypeError: cb.apply is not a function
> 
> 9 verbose stack     at <directory>.nvm/versions/node/v14.17.5/lib/node_modules/npx/node_modules/npm/node_modules/graceful-fs/polyfills.js:287:18
> 
> 9 verbose stack     at FSReqCallback.oncomplete (fs.js:193:5)
> 
> 10 verbose cwd <path-to-project>
> 
> 11 verbose Darwin 21.6.0
> 
> 12 verbose argv "<directory>/.nvm/versions/node/v14.17.5/bin/node" "<directory>/.nvm/versions/node/v14.17.5/lib/node_modules/npx/node_modules/npm/bin/npm-cli.js" "install" "vision-camera-plugin-builder@latest" "--global" "--prefix" "<directory>/.npm/_npx/73253" "--loglevel" "error" "--json"
> 
> 13 verbose node v14.17.5
> 
> 14 verbose npm  v5.1.0
> 
> 15 error cb.apply is not a function
> 
> 16 verbose exit [ 1, true ]
> 
> ```
> 
> 

It seems to be the problem with node 14, could you try with node 16?

--- a/generateIOS.rb
+++ b/generateIOS.rb
@@ -16,6 +16,10 @@ if lang == nil
   lang = "Swift"
 end
 
+def is_product_type_app_or_library(product_type)
+  return ["com.apple.product-type.application", "com.apple.product-type.library.static"].include? product_type
+end
+
 # Project reference
 project = Xcodeproj::Project.open(project_path)
 group = project.main_group
@@ -31,7 +35,7 @@ if lang == "Swift"
   is_bridging_header_created = false
   project.targets.each do |target|
     # Skip if target type is unit tests or app extension, etc.
-    if target.product_type == ("com.apple.product-type.application" || "com.apple.product-type.library.static")
+    if is_product_type_app_or_library(target.product_type)
 
       target.build_configurations.each do |config|
         if config.build_settings['SWIFT_OBJC_BRIDGING_HEADER'] == nil
@@ -129,7 +133,7 @@ if lang == "Swift"
 
   project.targets.each do |target|
     # Skip if target type is unit tests or app extension, etc.
-    if target.product_type == ("com.apple.product-type.application" || "com.apple.product-type.library.static")
+    if is_product_type_app_or_library(target.product_type)
       target.add_file_references([objc_impl_file_ref, swift_file_ref])
     end
   end
@@ -175,7 +179,7 @@ else
 
   project.targets.each do |target|
     # Skip if target type is unit tests or app extension, etc.
-    if target.product_type == ("com.apple.product-type.application" || "com.apple.product-type.library.static")
+    if is_product_type_app_or_library(target.product_type)
       target.add_file_references([objc_impl_file_ref])
     end
   end

